cst438_register/src/main/java/com/cst438/controller/CourseController.java

```diff
@@ -22,10 +24,22 @@
 	 */
 	@PutMapping("/course/{course_id}")
 	@Transactional
-	public void updateCourseGrades( @RequestBody CourseDTOG courseDTO, @PathVariable("course_id") int course_id) {
+	public void updateCourseGrades( @RequestBody CourseDTOG courseDTO, @PathVariable("course_id") int course_id) {	//courseDTOG, course_id
 		
 		//TODO  complete this method in homework 4
 		
+		CourseDTOG.GradeDTO gradeDTO = new CourseDTOG.GradeDTO();
```

Comment:
I don't think you need gradeDTO here. You should be able to get everything you need from the grade objects in courseDTO.

courseDTO.grades is a list of grade objects, so in each of those, there's the student information you need to fetch enrollment.

So in this method, there should be a loop that iterates through courseDTO.grades.